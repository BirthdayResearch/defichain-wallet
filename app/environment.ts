import * as Updates from 'expo-updates'

/**
 * Network supported in this environment
 */
export enum EnvironmentNetwork {
  LocalPlayground = 'Local Playground',
  RemotePlayground = 'Remote Playground',
  MainNet = 'MainNet',
  TestNet = 'TestNet'
}

export enum EnvironmentName {
  Production = 'Production',
  Preview = 'Preview',
  Development = 'Development',
}

interface Environment {
  name: EnvironmentName
  debug: boolean
  networks: EnvironmentNetwork[]
}

export const environments: Record<EnvironmentName, Environment> = {
  Production: {
    name: EnvironmentName.Production,
    debug: false,
    networks: [
      EnvironmentNetwork.RemotePlayground,
      EnvironmentNetwork.TestNet,
      EnvironmentNetwork.MainNet
    ]
  },
  Preview: {
    name: EnvironmentName.Preview,
    debug: true,
    networks: [
      EnvironmentNetwork.RemotePlayground,
      EnvironmentNetwork.TestNet,
      EnvironmentNetwork.MainNet
    ]
  },
  Development: {
    name: EnvironmentName.Development,
    debug: true,
    networks: [
      EnvironmentNetwork.LocalPlayground,
      EnvironmentNetwork.RemotePlayground
    ]
  }
}

/**
 * @return Environment of current Wallet, checked via expo release channels
 */
export function getEnvironment (): Environment {
  const channel = Updates.releaseChannel

  if (channel === 'production') {
    return environments[EnvironmentName.Production]
  }

  if (channel.startsWith('pr-preview-') || channel === 'prerelease') {
    return environments[EnvironmentName.Preview]
  }

  return environments[EnvironmentName.Development]
}

/**
 * @param {EnvironmentNetwork} network to check if it is a playground network
 */
export function isPlayground (network: EnvironmentNetwork): boolean {
  return [
    EnvironmentNetwork.LocalPlayground,
    EnvironmentNetwork.RemotePlayground
  ].includes(network)
}
