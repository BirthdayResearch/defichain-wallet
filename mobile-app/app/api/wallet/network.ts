import { MainNet, Network, RegTest, TestNet } from '@defichain/jellyfish-network'
import { Bip32Options } from '@defichain/jellyfish-wallet-mnemonic'
import { EnvironmentNetwork } from '@environment'

export function getJellyfishNetwork (network: EnvironmentNetwork): Network {
  switch (network) {
    case EnvironmentNetwork.MainNet:
      return MainNet
    case EnvironmentNetwork.TestNet:
      return TestNet
    case EnvironmentNetwork.LocalPlayground:
    case EnvironmentNetwork.RemotePlayground:
      return RegTest
  }
}

export function getBip32Option (envNetwork: EnvironmentNetwork): Bip32Options {
  const network = getJellyfishNetwork(envNetwork)
  return {
    bip32: {
      public: network.bip32.publicPrefix,
      private: network.bip32.privatePrefix
    },
    wif: network.wifPrefix
  }
}
