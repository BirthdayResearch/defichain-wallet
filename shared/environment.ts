/**
 * Network supported in this environment
 */
export enum EnvironmentNetwork {
  LocalPlayground = 'Local',
  RemotePlayground = 'Playground',
  MainNet = 'MainNet',
  TestNet = 'TestNet'
}

export enum EnvironmentName {
  Production = 'Production',
  Preview = 'Preview',
  Development = 'Development',
}

interface Environment {
  name: EnvironmentName
  debug: boolean
  networks: EnvironmentNetwork[]
}

export const environments: Record<EnvironmentName, Environment> = {
  Production: {
    name: EnvironmentName.Production,
    debug: false,
    networks: [
      EnvironmentNetwork.MainNet,
      EnvironmentNetwork.TestNet,
      EnvironmentNetwork.RemotePlayground
    ]
  },
  Preview: {
    name: EnvironmentName.Preview,
    debug: true,
    networks: [
      EnvironmentNetwork.RemotePlayground,
      EnvironmentNetwork.TestNet,
      EnvironmentNetwork.MainNet
    ]
  },
  Development: {
    name: EnvironmentName.Development,
    debug: true,
    networks: [
      EnvironmentNetwork.LocalPlayground,
      EnvironmentNetwork.RemotePlayground,
      EnvironmentNetwork.TestNet,
      EnvironmentNetwork.MainNet
    ]
  }
}

/**
 * @return Environment of current Wallet, checked via expo release channels
 */
export function getEnvironment (channel: string): Environment {
  if (channel === 'production') {
    return environments[EnvironmentName.Production]
  }

  if (channel.startsWith('pr-preview-') || channel === 'prerelease') {
    return environments[EnvironmentName.Preview]
  }

  return environments[EnvironmentName.Development]
}

/**
 * @param {EnvironmentNetwork} network to check if it is a playground network
 */
export function isPlayground (network: EnvironmentNetwork): boolean {
  return [
    EnvironmentNetwork.LocalPlayground,
    EnvironmentNetwork.RemotePlayground
  ].includes(network)
}
