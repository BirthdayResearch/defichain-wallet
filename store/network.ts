import { createSlice, PayloadAction } from '@reduxjs/toolkit'
import { createSelector } from 'reselect'
import { RootState } from './index'

export type NetworkName = 'mainnet' | 'testnet' | 'regtest' | 'playground'

export interface PlaygroundApiState {
  url: string
  environment: 'localhost' | 'ocean'
}

export interface WhaleApiState {
  url: string
  network: NetworkName
}

interface NetworkState {
  name?: NetworkName
  whale?: WhaleApiState
  playground?: PlaygroundApiState
}

const initialState: NetworkState = {}

export const network = createSlice({
  name: 'network',
  initialState,
  reducers: {
    setWhale: (state, action: PayloadAction<WhaleApiState>) => {
      state.whale = action.payload
      state.name = action.payload.network
    },
    setPlayground: (state, action: PayloadAction<PlaygroundApiState>) => {
      state.playground = action.payload
    }
  }
})

export const refreshIntervalSelector = createSelector((state: RootState) => state.network.name, (name: NetworkName | undefined) => {
  switch (name) {
    case 'playground':
    case 'regtest':
    default:
      return 3000
    case 'testnet':
      return 15000
    case 'mainnet':
      return 30000
  }
}
)
