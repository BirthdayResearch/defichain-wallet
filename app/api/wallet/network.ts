import { MainNet, Network, RegTest, TestNet } from '@defichain/jellyfish-network'
import { EnvironmentNetwork } from '../../environment'

export function getJellyfishNetwork (network: EnvironmentNetwork): Network {
  switch (network) {
    case EnvironmentNetwork.MainNet:
      return MainNet
    case EnvironmentNetwork.TestNet:
      return TestNet
    case EnvironmentNetwork.LocalPlayground:
    case EnvironmentNetwork.RemotePlayground:
      return RegTest
  }
}
