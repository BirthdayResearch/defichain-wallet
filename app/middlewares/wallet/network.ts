import { MainNet, Network, RegTest, TestNet } from '@defichain/jellyfish-network'
import { EnvironmentNetwork } from '../../environment'
import { getNetwork } from '../storage'

export async function getNetworkOptions (): Promise<Network> {
  const network = await getNetwork()

  switch (network) {
    case EnvironmentNetwork.MainNet:
      return MainNet
    case EnvironmentNetwork.TestNet:
      return TestNet
    case EnvironmentNetwork.LocalPlayground:
    case EnvironmentNetwork.RemotePlayground:
      return RegTest
  }
}
