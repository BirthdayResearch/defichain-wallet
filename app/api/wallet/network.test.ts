import { EnvironmentNetwork } from "../../environment";
import { getBip32Option, getTxURLByNetwork } from "./network";

it('should resolve bip32 option for MainNet', () => {
  const bip32Options = getBip32Option(EnvironmentNetwork.MainNet)
  expect(bip32Options).toMatchSnapshot()
})

it('should resolve bip32 option for TestNet', () => {
  const bip32Options = getBip32Option(EnvironmentNetwork.TestNet)
  expect(bip32Options).toMatchSnapshot()
})

it('should resolve bip32 option for LocalPlayground', () => {
  const bip32Options = getBip32Option(EnvironmentNetwork.LocalPlayground)
  expect(bip32Options).toMatchSnapshot()
})

it('should resolve bip32 option for RemotePlayground', () => {
  const bip32Options = getBip32Option(EnvironmentNetwork.RemotePlayground)
  expect(bip32Options).toMatchSnapshot()
})

it('should return correct tx url for MainNet', () => {
	const txUrl = getTxURLByNetwork(EnvironmentNetwork.MainNet, '123')
	expect(txUrl).toMatchSnapshot()
})

it('should return correct tx url for TestNet', () => {
	const txUrl = getTxURLByNetwork(EnvironmentNetwork.TestNet, '123')
	expect(txUrl).toMatchSnapshot()
})

it('should return correct tx url for LocalPlayground', () => {
	const txUrl = getTxURLByNetwork(EnvironmentNetwork.LocalPlayground, '123')
	expect(txUrl).toMatchSnapshot()
})

it('should return correct tx url for RemotePlayground', () => {
	const txUrl = getTxURLByNetwork(EnvironmentNetwork.RemotePlayground, '123')
	expect(txUrl).toMatchSnapshot()
})
