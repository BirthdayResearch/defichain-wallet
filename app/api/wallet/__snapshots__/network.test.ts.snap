// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should resolve bip32 option for LocalPlayground 1`] = `
Object {
  "bip32": Object {
    "private": 70615956,
    "public": 70617039,
  },
  "wif": 239,
}
`;

exports[`should resolve bip32 option for MainNet 1`] = `
Object {
  "bip32": Object {
    "private": 76066276,
    "public": 76067358,
  },
  "wif": 128,
}
`;

exports[`should resolve bip32 option for RemotePlayground 1`] = `
Object {
  "bip32": Object {
    "private": 70615956,
    "public": 70617039,
  },
  "wif": 239,
}
`;

exports[`should resolve bip32 option for TestNet 1`] = `
Object {
  "bip32": Object {
    "private": 70615956,
    "public": 70617039,
  },
  "wif": 239,
}
`;
